### ISSUE TITLE
2.4.11 Focus Visible (Enhanced) - what counts as "bounding rectangle"/"bounding box"

### ISSUE THREAD

**Issue by patrickhlauke** [w3c/wcag#1075]
there is no clear definition of what "bounding rectangle" (and later in the understanding doc, "bounding box") refers to. is it the "clickable"/"active" area of a control? or its visual apparent bounding box?

thinking of situations where the clickable area of a control has been made quite large (perhaps to pass some target size requirement, or just to make it more touch friendly) by using ample padding that goes beyond the "apparent" confines of a control ... is the bounding box the visually apparent box, or the actual clickable/tappable box? if the latter, this would introduce an odd conundrum that two controls that *appear* the same may end up needing differently sized focus indicators, purely because one also made an extra effort to have a more generous invisible "slop" area around it.

**Comment by detlevhfischer** [comment]
Can we point to the latest version of 2.4.11 Focus Visible (Enhanced) from here?

**Comment by patrickhlauke** [comment]
which one, this? https://w3c.github.io/wcag/guidelines/22/#focus-visible-enhanced

**Comment by patrickhlauke** [comment]
(which i see swaps out "bounding box" for "bounding rectangle")

**Comment by detlevhfischer** [comment]
In order to avoid the conundrum you are pointing out I would assume (and have assumed, before you raised the question) that the bounding rectangle would be based on the visible target (as a minimum requirement).

**Comment by patrickhlauke** [comment]
but the problem is...what is the visible target? that's not always clear. what if i have a giant button with no border/background, that only contains a single "GO >" bit of text with a LOT of padding around it, again with no border/background. is the assumed visible target then just the text? (probably yes, but this should be defined/explained)

**Comment by alastc** [comment]
I can see two points here:

1. What should the bounding box be applied to, the visible component or the actual element? (If they differ.)
2. Does the term 'bounding box' need to be defined?

For the first we have #1004 

Does "bounding box" need to be defined? We could do. @awkawk and I discussed that and it seems to be a common term in apps (e.g. design programs), computer science and geometry. The google results for bounding box are reasonable, but if people don't get it I'd propose something like:

"The smallest rectangle that can be drawn around a shape."

**Comment by patrickhlauke** [comment]
the current problem is that since case 1 is not explicitly mentioned currently in the understanding, this brought up the need for 2. adding an explicit mention/explanation for 1 will likely obviate the need for 2. (saying that it's "The smallest rectangle that can be drawn around a shape" would be pointless if you then don't define what "shape" is actually to be considered, which is the point of 1)

**Comment by alastc** [comment]
Sure, I'm just saying that this is (effectively) a duplicate then, we should resolve 1004.

**Comment by alastc** [comment]
Hi @patrickhlauke,

The latest version of the sizing bullet ([option 3 here](https://docs.google.com/document/d/1F83m-HXRkXz1QCF6_QNtQGzIULugMiFSLgtqDRgKA-s/edit#)) is:

> Minimum area: The focus indication area is greater than or equal to a 1px border of the focused control.

That slightly increases the requirement to 1px along the border of the whole control rather than the longest-sides. That has the benefit of being simpler to understand and not needing the term 'bounding box'. It is also explicit about the control being the 'thing' for the sizing.

Does that help?

**Comment by scha14** [comment]
@patrickhlauke given that [option 3 in the next version proposal] (https://docs.google.com/document/d/1F83m-HXRkXz1QCF6_QNtQGzIULugMiFSLgtqDRgKA-s/edit#) has an overall border instead of the term bounding rectangle or associated complexity, is it okay to close this issue?

**Comment by scha14** [comment]
Hi @patrickhlauke closing this issue for now, please feel free to reopen in case there are unaddressed concerns. Thank you!

**Comment by 36degrees** [comment]
Given:

> The working group is interested in feedback about the minimum area metric, and if there are unusual scenarios where visible indicators are caught by the wording.

I thought it'd be worth raising this as a possible 'unusual scenario'.

It seems very relevant to this issue, but I'd be happy to write this up as a separate issue if you'd prefer rather than re-opening this one.

---

I've been looking at the WCAG 2.2 focus appearance criteria would apply to the [accordion component from the GOV.UK Design System](https://design-system.service.gov.uk/components/accordion/):

![Controls for an accordion component with two sections. The section headings are aligned to the left, with a plus symbol on the far right side of each control. Each section is divided by a border.](https://user-images.githubusercontent.com/121939/90774675-bf22a000-e2ef-11ea-8249-b674f5ff3b50.png)

The hit area to expand a section is the entire area between the top and bottom border:

![Highlighted version of the previous diagram showing the hit area for each section, which is the entire area between the top and bottom border for the section.](https://user-images.githubusercontent.com/121939/90774987-2f312600-e2f0-11ea-9bab-d3642fd562ac.png)

The 'Understanding Success Criterion 2.4.11: Focus Appearance (Minimum)' document has the note:

> "If the control has a visible boundary smaller than the hit area, the size measure is taken from the visible boundary."

However in this instance we seem to have two visible indicators within the same hit area – the section header and the '+' on the right hand side.

How should the size measure be calculated in this instance? There seems to be a few options…

### Drawing the smallest possible area that covers both visible boundaries:

This results in a very large minimum focus indication area.

![Annotated version of the first diagram showing a potential size measure taken from the smallest possible area that covers both the heading and the plus symbol.](https://user-images.githubusercontent.com/121939/90775850-40c6fd80-e2f1-11ea-95ae-6897465825ac.png)

945px + 945px + 30px + 30px = 1950px

### Summing the visible area of the two elements:

![Annotated version of the first diagram showing a potential size measures taken as a border around the heading plus a separate border around the plus symbol.](https://user-images.githubusercontent.com/121939/90775909-563c2780-e2f1-11ea-8daa-10ae02d91f23.png)

104px + 104px + 30px + 30px + 16px + 16px + 16px + 16px = 332px

### Only considering the heading

I think this is effectively treating the '+' as a supplementary element outside of the control but within the control's hit area?

![Annotated version of the first diagram showing a potential size measures taken as a border around the heading only.](https://user-images.githubusercontent.com/121939/90775975-694ef780-e2f1-11ea-92d5-f3c490a2c454.png)

104px + 104px + 30px + 30px = 268px

---

For context, this is our current focus state:

![Accordion with the first section focused. The section heading has a yellow background and a 4px black border across the bottom of the heading, like an underline.](https://user-images.githubusercontent.com/121939/90775231-820add80-e2f0-11ea-86f2-605647a59e91.png)

It has a focus indication area of 104px × 4 = 416px (the yellow is not treated as part of the focus indication area, as it does not meet the change of contrast criteria). I would consider it to be visible.
